contracts/zero-ex/contracts/src/features/multiplex/MultiplexFeature.sol

```diff
@@ -544,7 +544,11 @@ contract MultiplexFeature is
                 // is executed, so we the target is the address encoded
                 // in the subcall data.
                 (target, ) = abi.decode(subcall.data, (address, bytes));
-            } else if (subcall.id == MultiplexSubcall.UniswapV3 || subcall.id == MultiplexSubcall.BatchSell) {
+            } else if (
+                subcall.id == MultiplexSubcall.UniswapV3 ||
+                subcall.id == MultiplexSubcall.BatchSell ||
+                subcall.id == MultiplexSubcall.OTC
```

Comment:
The logic in `_multiHopSellOtcOrder` that uses `state.hopIndex` to compute the address to pull funds from is not compatible with `params.useSelfBalance`. It would be a pretty quick fix, but as it stands I would expect it not to work now that `OTC` is handled the same as `UniswapV3` and `BatchSell`

Reply:
https://github.com/0xProject/protocol/pull/665/commits/172822044b2c21c221e63d1a1fe9d1af9b642d53